lib/puppet/context.rb

```diff
@@ -40,6 +40,7 @@ def pop
     if @stack[-1][0] == 0
       raise(StackUnderflow, "Attempted to pop, but already at root of the context stack.")
     else
+      @id -= 1
```

Comment:
The `@id` field isn't directly exposed and you can only really observe it by stepping into the class via the debugger. I assumed the id was meant to indicate the context depth but it could be a unique identifier. Is that more helpful, identifying the context or stack depth?

But since this isn't exposed anywhere I can drop this commit and we can pretend this question never came up.